frontend/src/components/QueryBuilder/QueryBuilder.tsx

```diff
@@ -1746,35 +1417,34 @@ function QueryBuilder<T extends SearchParamsInput | ErrorSearchParamsInput>(
 	}
 	const addRule = useCallback(
 		(rule: RuleProps) => {
-			setRules([...rules, rule])
+			setRulesImpl([...rules, rule])
 			setCurrentRule(undefined)
 		},
-		[rules],
+		[rules, setRulesImpl],
 	)
 	const removeRule = useCallback(
 		(targetRule: RuleProps) =>
-			setRules(rules.filter((rule) => rule !== targetRule)),
-		[rules],
+			setRulesImpl(rules.filter((rule) => rule !== targetRule)),
+		[rules, setRulesImpl],
+	)
+	const updateRule = useCallback(
+		(targetRule: RuleProps, newProps: any) => {
+			setRulesImpl(
+				rules.map((rule) =>
+					rule !== targetRule ? rule : { ...rule, ...newProps },
+				),
+			)
+		},
+		[rules, setRulesImpl],
 	)
-	const updateRule = (targetRule: RuleProps, newProps: any) => {
-		setRulesImpl((currentRules) =>
-			currentRules.map((rule) =>
-				rule !== targetRule ? rule : { ...rule, ...newProps },
-			),
-		)
-	}
 
 	const timeRangeRule = useMemo<RuleProps>(() => {
 		const timeRange = rules.find(
 			(rule) => rule.field?.value === timeRangeField.value,
-		)
-		if (!timeRange) {
-			addRule(defaultTimeRangeRule)
-			return defaultTimeRangeRule
-		}
+		)! // ZANETODO: can we enforce this?
```

Comment:
we should separate the timeRangeRule from the rules eventually since it is required